tests/Elasticsearch.Extensions.Logging.IntegrationTests/LoggingTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Elastic.CommonSchema;
+using Elastic.Elasticsearch.Xunit;
+using Elastic.Elasticsearch.Xunit.XunitPlumbing;
+using Elasticsearch.Net;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Nest;
+using Xunit;
+using LogLevel = Microsoft.Extensions.Logging.LogLevel;
+
+namespace Elasticsearch.Extensions.Logging.IntegrationTests
+{
+	public class LoggingTests : IClusterFixture<LoggingCluster>
+	{
+		private ElasticClient Client { get; }
+
+		public LoggingTests(LoggingCluster cluster) =>
+			Client = cluster.GetOrAddClient(c =>
+			{
+				var nodes = cluster.NodesUris();
+				var connectionPool = new StaticConnectionPool(nodes);
+				var settings = new ConnectionSettings(connectionPool)
+					.EnableDebugMode();
+				return new ElasticClient(settings);
+			});
+
+		private IDisposable CreateLogger(out ILogger logger, out string indexPrefix)
+		{
+			var pre = $"logs-{Guid.NewGuid().ToString("N").ToLowerInvariant().Substring(0, 6)}";
+			var options = new ConfigureOptions<ElasticsearchLoggerOptions>(
+				o =>
+				{
+					o.Index = $"{pre}-{{0:yyyy.MM.dd}}";
+					var nodes = Client.ConnectionSettings.ConnectionPool.Nodes.Select(n => n.Uri).ToArray();
+					o.ShipTo = new ShipTo(nodes, ConnectionPoolType.Static);
+					
+				});
+
+			var optionsFactory = new OptionsFactory<ElasticsearchLoggerOptions>(
+				new []{ options }, Enumerable.Empty<IPostConfigureOptions<ElasticsearchLoggerOptions>>());
+			var optionsMonitor = new OptionsMonitor<ElasticsearchLoggerOptions>(
+				optionsFactory, Enumerable.Empty<IOptionsChangeTokenSource<ElasticsearchLoggerOptions>>(), new OptionsCache<ElasticsearchLoggerOptions>());
+			var loggerFactory = new LoggerFactory(
+				new[] { new ElasticsearchLoggerProvider(optionsMonitor) }, new LoggerFilterOptions { MinLevel = LogLevel.Information });
+			logger = loggerFactory.CreateLogger<ElasticsearchLogger>();
+			indexPrefix = pre;
+			return loggerFactory;
+		}
+
+		[Fact]
+		public async Task LogsEndUpInCluster()
+		{
+			using var _ = CreateLogger(out var logger, out var indexPrefix);
+
+			logger.LogError("an error occured");
+
+			// TODO make sure we can await something here on ElasticsearchDataShipper
+			await Task.Delay(TimeSpan.FromSeconds(5));
```

Comment:
Could the logger expose a property/event to ascertain that it has sent or is sending events?

Reply:
That was my intend but I knew the shipper had to be rewritten so I left the TODO :) 

Once #99 lands we can expose more callbacks and stats